### PR TITLE
Fixed bug related to limits of abs

### LINKED ISSUE #26513: Wrong limit result for Abs((-n/(n+1))**n)
```python
import sympy

from sympy import S, Symbol, limit, oo

print('sympy version:', sympy.__version__)

n = Symbol('n', real=True)

expr = (n/(n+1))**n
print(limit(expr, n, oo))  # Success

expr = (-n/(n+1))**n
print(limit(expr, n, oo))  # Wrong

expr = Abs((-n/(n+1))**n)
print(limit(expr, n, oo))  # Wrong
```

The result:
```python
sympy version: 1.12
exp(-1)
0
0
```

### PR DESCRIPTION
**Pull request by arnabnandikgp**
#### References to other Issues or PRs

Fixes #26513 
#### Brief description of what is fixed or changed
sign function in gruntz gave erroneous results for negative logarithmic arguments. Refactored `set_signs` in limits to let pass expressions returning NotImplementedError for gruntz to operate directly on abs functions . Made this separate PR for this issue as suggested in  #26749.

#### Release Notes

* series
  * Fixed `sign` in gruntz for better handling of terms having negative bases in power expressions with symbolic exponents.

### CODE REVIEW

File: sympy/series/tests/test_limits.py [hunk 1]
```diff
@@ -1414,6 +1414,11 @@ def test_issue_26250():
     assert limit(e1/e2, x, 0) == -S(1)/8
 
 
+def test_issue_26513():
+    assert limit(abs((-n/(n+1))**n), n ,oo) == exp(-1)
+    raises (NotImplementedError, lambda: limit((-n/(n+1))**n, n, oo))
+
```
Comment (anutosh491) [inline sympy/series/tests/test_limits.py:1420]: I guess we should also add the tests which already work on master as control .  Can add all of these ig
```
    assert limit((x/(x + 1))**x, x, oo) == exp(-1)
    assert limit((-x/(x + 1))**x, x, oo) == Limit((-x/(x + 1))**x, x, oo)
    assert limit(abs((-x/(x + 1))**x), x, oo) == exp(-1)
    
```

Comment (arnabnandikgp) [reply sympy/series/tests/test_limits.py:1421]: we cannot return an unevaluated limit for the second example u have stated as this roots from here
https://github.com/sympy/sympy/blob/95a8175eb35ca7c8d84afadafb8df4bfb93fd1d2/sympy/series/gruntz.py#L650-L653

Comment (arnabnandikgp) [reply sympy/series/tests/test_limits.py:1421]: that was the whole point of the past pr not getting merged as we would then have to catch it here,
https://github.com/sympy/sympy/blob/95a8175eb35ca7c8d84afadafb8df4bfb93fd1d2/sympy/series/limits.py#L376-L381

Comment (anutosh491) [reply sympy/series/tests/test_limits.py:1421]: Yeah yeah was just pasting the examples here. Results might vary

File: sympy/series/limits.py [hunk 2]
```diff
@@ -273,16 +273,20 @@ def set_signs(expr):
...
-                                S.NegativeOne if sign_flag else S.Pi)
-                    elif (sig > 0) == True:
-                        return (expr.args[0] if abs_flag else
-                                S.One if sign_flag else S.Zero)
+                try:
+                    sig = limit(expr.args[0], z, z0, dir)
+                    if sig.is_zero:
+                        sig = limit(1/expr.args[0], z, z0, dir)
+                except NotImplementedError:
+                    pass
```
Comment (anutosh491) [inline sympy/series/limits.py:281]: Just a doubt. Are we sure we want to use pass here ?
Do we have any other options and like what result do we get if we don't have a try except block here ?

**Comment by arnabnandikgp** [comment]
@anutosh491 I think this can go in, these changes were approved by y'all in the previous pr, can you please look into this.

**Review by anutosh491 — APPROVED** [review]
cc @arnabnandikgp I think this is good and ready. If no concerns are raised I shall merge it by today

**Pull request merged**